lib/src/main/kotlin/WeatherSensorService/Internal/MockWeatherSensorReader.kt

```diff
@@ -57,11 +61,15 @@ class MockWeatherSensorReader: WeatherSensorReaderType {
      private fun reportSensorReadings() {
          flow.tryEmit(
              WeatherSensorReading(
-                Random.nextDouble(-40.0, 40.0),
-                Random.nextDouble(0.0,100.0),
-                Random.nextDouble(95.0,105.0),
+                GenerateTemperatureReadings((waveLocation/100.0),Constants.TEMPERATURE_RANGE),
+                GenerateTemperatureReadings((waveLocation/100.0),Constants.HUMIDITY_RANGE),
+                GenerateTemperatureReadings((waveLocation/100.0),Constants.PRESSURE_RANGE),
                 LocalDateTime.now()
              )
          )
+         if ( waveLocation === 100) {
+             waveLocation = 1
+            }
+        waveLocation++
```

Comment:
- Indentation is off on the closing }
- `===` checks referential equality. I'm not sure whether Int is a reference type in Kotlin, but even if it is, we only care whether there is structural equality here, so this should be using `==`.